tests/bearlib/abstractions/LinterTest.py

```diff
@@ -0,0 +1,711 @@
+import os
+import sys
+import unittest
+from unittest.mock import ANY, Mock
+
+from coalib.bearlib.abstractions.Linter import Linter
+from coalib.results.Diff import Diff
+from coalib.results.Result import Result
+from coalib.results.RESULT_SEVERITY import RESULT_SEVERITY
+from coalib.settings.Section import Section
+
+
+def get_testfile_name(name):
+    """
+    Gets the full path to a testfile inside ``linter_test_files`` directory.
+
+    :param name: The filename of the testfile to get the full path for.
+    :return:     The full path to given testfile name.
+    """
+    return os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                        "linter_test_files",
+                        name)
+
+
+class LinterComponentTest(unittest.TestCase):
+
+    # Using `object` instead of an empty class results in inheritance problems
+    # inside the linter decorator.
+    class EmptyTestLinter:
+        pass
+
+    class ManualProcessingTestLinter:
+
+        def process_output(self, *args, **kwargs):
+            pass
+
+    def setUp(self):
+        self.section = Section("TEST_SECTION")
+
+    def test_decorator_invalid_parameters(self):
+        with self.assertRaises(ValueError) as cm:
+            Linter("some-executable", invalid_arg=88, ABC=2000)
+        self.assertEqual(
+            str(cm.exception),
+            "Invalid keyword arguments provided: 'ABC', 'invalid_arg'")
+
+        with self.assertRaises(ValueError) as cm:
+            Linter("some-executable", diff_severity=RESULT_SEVERITY.MAJOR)
+        self.assertEqual(str(cm.exception),
+                         "Invalid keyword arguments provided: 'diff_severity'")
```

Comment:
can you make each of those an own function? They all test a different invalidity of params, right? Let's seperate them.